clis/teliod/src/daemon.rs

```diff
@@ -0,0 +1,242 @@
+                    if let Err(e) = response_tx_channel
+                        .blocking_send(CommandResponse::StatusReport(status_report))
+                    {
+                        error!("Telio task failed sending response: {}", e)
+                    }
+                }
+                TelioTaskCmd::Quit => {
+                    telio.stop();
+                    break;
+                }
+            }
+        }
+    }
+    Ok(())
+}
+
+fn task_retrieve_meshmap(
+    device_identity: Arc<DeviceIdentity>,
+    auth_token: Arc<String>,
+    tx: mpsc::Sender<TelioTaskCmd>,
+) {
+    tokio::spawn(async move {
+        let result = get_meshmap_from_server(device_identity, auth_token).await;
+        match result {
+            Ok(meshmap) => {
+                trace!("Meshmap {:#?}", meshmap);
+                #[allow(mpsc_blocking_send)]
+                if let Err(e) = tx.send(TelioTaskCmd::UpdateMeshmap(meshmap)).await {
+                    error!("Unable to send meshmap due to {e}");
+                }
+            }
+            Err(e) => error!("Getting meshmap failed due to {e}"),
+        }
+    });
+}
+
+fn start_telio(
+    telio: &mut Device,
+    private_key: SecretKey,
+    interface_name: &str,
+) -> Result<(), DeviceError> {
+    telio.start(&DeviceConfig {
+        private_key,
+        name: Some(interface_name.to_owned()),
+        adapter: AdapterType::BoringTun,
+        ..Default::default()
+    })?;
+
+    debug!("started telio with {:?}...", AdapterType::BoringTun,);
+    Ok(())
+}
+
+pub async fn daemon_event_loop(config: TeliodDaemonConfig) -> Result<(), TeliodError> {
+    let (non_blocking_writer, _tracing_worker_guard) =
+        tracing_appender::non_blocking(fs::File::create(&config.log_file_path)?);
+    tracing_subscriber::fmt()
+        .with_max_level(config.log_level)
+        .with_writer(non_blocking_writer)
+        .with_ansi(false)
+        .with_line_number(true)
+        .with_level(true)
+        .init();
+
+    debug!("started with config: {config:?}");
+
+    let socket = DaemonSocket::new(&DaemonSocket::get_ipc_socket_path()?)?;
+
+    let nc = NotificationCenter::new(&config).await?;
+
+    // Tx is unused here, but this channel can be used to communicate with the
+    // telio task
+    let (tx, rx) = mpsc::channel(10);
+    let (response_tx, response_rx) = mpsc::channel(10);
+    let mut cmd_listener = CommandListener::new(socket, tx.clone(), response_rx);
+
+    // TODO: This if condition and ::default call is temporary to be removed later
+    // on when we have proper integration tests with core API.
+    // This is to not look for tokens in a test environment right now as the values
+    // are dummy and program will not run as it expects real tokens.
+    let mut identity = DeviceIdentity::default();
+    if !config
+        .authentication_token
+        .eq("aaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaa")
```

Comment:
nit: make it a EMPTY_TOKEN const?

Reply:
Same as above, but good idea, I can quickly refactor this too

Reply:
Moved to `EMPTY_TOKEN`